### ISSUE TITLE
Add ability to toggle between front and rear camera 

### ISSUE THREAD

**Issue by GoogleCodeExporter** [BinaryBlue/imsdroid#84]
```
What steps will reproduce the problem?
1.For Samsung Galaxy, HTC EVO and other phones with a front facing camera, it 
would be nice to offer a setting that allows either real-time toggling of the 
camera or prior to call-setup.
2.
3.

What is the expected output? What do you see instead?

Currently it defaults to front camera for these phones

For AT&T Captivate, which is a Galaxy platform based phone which does not have 
a front camera, video calling does not work since the code does not
successfully use the rear camera. Having an explicit toggle camera setting may 
help this
What version of the product are you using? On what operating system?


Please provide any additional information below.


```

Original issue reported on code.google.com by `saket...@gmail.com` on 27 Oct 2010 at 2:07

**Comment by GoogleCodeExporter** [comment]
Original comment by `boss...@yahoo.fr` on 27 Oct 2010 at 7:24
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in SVN r316.
Go to Options->General check or uncheck "Enable Front Facing Camera" checkbox.
```

Original comment by `boss...@yahoo.fr` on 28 Oct 2010 at 10:04

**Comment by GoogleCodeExporter** [comment]
```
Verified that rear camera is working on the AT&T Captivate (Samsung Galaxy 
without front facing camera) with this new option.
```

Original comment by `saket...@gmail.com` on 30 Oct 2010 at 11:56

**Comment by GoogleCodeExporter** [comment]
```
Could I mark the issue as solved?
```

Original comment by `boss...@yahoo.fr` on 2 Nov 2010 at 12:41

**Comment by GoogleCodeExporter** [comment]
```
Yes you can mark it as solved. A possible enhancement is to add a toggle camera 
button in the call window itself instead of the settings window -- so that 
during an active call, a user can toggle back and forth between the front and 
rear camera, if both exist.
```

Original comment by `saket...@gmail.com` on 3 Nov 2010 at 11:58

**Comment by GoogleCodeExporter** [comment]
```
i was working on the change suggested by saket424. I think that would be a 
great addition. We can re-use answer button (it is grayed out once the call is 
in progress) to toggle between cameras. I just need a good icon for camera 
toggle. will let you know when i have it working.
```

Original comment by `alex9...@gmail.com` on 5 Nov 2010 at 1:11

**Comment by GoogleCodeExporter** [comment]
```
Latest SVN version fixes the issue (Thanks to Alex)
```

Original comment by `boss...@yahoo.fr` on 16 Nov 2010 at 12:16

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 1.1.327
```

Original comment by `boss...@yahoo.fr` on 22 Nov 2010 at 6:18

**Comment by GoogleCodeExporter** [comment]
Original comment by `boss...@yahoo.fr` on 22 Nov 2010 at 6:19
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
i think one item still remains to close out this issue completely. the icon on 
the menu is still the ("Answer") icon as in original version. It would be good 
to have another icon for front/back camera toggle.
```

Original comment by `alex9...@gmail.com` on 22 Nov 2010 at 11:51

**Comment by GoogleCodeExporter** [comment]
Original comment by `boss...@yahoo.fr` on 22 Nov 2010 at 7:47
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
i am using a motorola device in which front camera is not present but back is 
there
so is it possible to make a video call by this device or not...
another question is that if i want to make a video  call by emulator by using 
the webcam is it possible to do or not? if it is possible by using the emulator 
then HOW?
please HELP me
thanks

```

Original comment by `sharad.s...@gmail.com` on 24 Nov 2010 at 11:44

**Comment by GoogleCodeExporter** [comment]
```
I will answer the first question. Yes it is possible. However you will have
only 1 way video. You can choose if you want to see the remote or the remote
will see you.because you only have back camera you can't do both because of
you hardware. Flip will only need to be enabled if you decide to send video
from your device.

front
```

Original comment by `alex9...@gmail.com` on 24 Nov 2010 at 12:56

**Comment by GoogleCodeExporter** [comment]
```
@sharad.sgit

1. Off course you can make 2-way video call without front facing camera. The 
software will use the back camera. What happen when you press "Send Video"?
2. The emulator cannot use the desktop camera but it has a fake camera. To 
enable the camera on the emulator you must start it with an sdcard. It's not a 
good idea to test video feature on the emulator and you will probably have many 
problems (slow, crash, ...)
3. There is no relation between flip and sending video. Flip is used to rotate 
(90°) the outgoing stream.
```

Original comment by `boss...@yahoo.fr` on 24 Nov 2010 at 9:00

**Comment by GoogleCodeExporter** [comment]
```
Thanks
Mr.
boss...@yahoo.fr
One more question is it only possible to make a Video cal from imsdroid to 
imsdroid
means only on Android OS or we can make a Video cal to others OS softphone 
please reply  
```

Original comment by `sharad.s...@gmail.com` on 25 Nov 2010 at 5:35

**Comment by GoogleCodeExporter** [comment]
```
IMSDroid is based on standards (SIP, SDP, RTP, ...) which means that you can 
call any other device running on any OS as far as it's a SIP client.
For example, on iOS (iPhone, iPad and iPod Touch) you can make video call with 
iDoubs: http://code.google.com/p/idoubs/
On PC: Linphone, Ekiga, SIP Communicator, Boghe, Bria, Eyebeam, ...
```

Original comment by `boss...@yahoo.fr` on 25 Nov 2010 at 5:41

**Comment by GoogleCodeExporter** [comment]
```
Fixed by "imsdroid-2.0.419-preview".
```

Original comment by `boss...@yahoo.fr` on 5 Jun 2011 at 5:20

**Comment by GoogleCodeExporter** [comment]
Original comment by `boss...@yahoo.fr` on 5 Jun 2011 at 5:37
- Changed state: **Fixed**
